diff --git a/Buildings/Templates/AirHandlersFans/VAVMultiZone.mo b/Buildings/Templates/AirHandlersFans/VAVMultiZone.mo
index 50225a86ef..75a2977eb2 100644
--- a/Buildings/Templates/AirHandlersFans/VAVMultiZone.mo
+++ b/Buildings/Templates/AirHandlersFans/VAVMultiZone.mo
@@ -406,6 +406,7 @@ equation
   connect(fanSupBlo.port_b, coiHeaPre.port_a)
     annotation (Line(points={{-30,-200},{10,-200}}, color={0,127,255}));
   annotation (
+    __LinkageTemplate=true,
     defaultComponentName="VAV",
     Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(
     coordinateSystem(preserveAspectRatio=false, extent={{-300,-280},{300,280}}),
diff --git a/Buildings/Templates/AirHandlersFans/package.mo b/Buildings/Templates/AirHandlersFans/package.mo
index ecce01041c..018737416f 100644
--- a/Buildings/Templates/AirHandlersFans/package.mo
+++ b/Buildings/Templates/AirHandlersFans/package.mo
@@ -2,7 +2,9 @@ within Buildings.Templates;
 package AirHandlersFans "Air handlers and fans"
   extends Modelica.Icons.Package;
 
-  annotation (Documentation(info="<html>
+  annotation (
+    __LinkageTemplate=true,
+    Documentation(info="<html>
 <p>
 This package contains templates for air handlers and fans.
 </p>
diff --git a/Buildings/Templates/ZoneEquipment/VAVBoxCoolingOnly.mo b/Buildings/Templates/ZoneEquipment/VAVBoxCoolingOnly.mo
index 6d3893617b..986c7fa3f8 100644
--- a/Buildings/Templates/ZoneEquipment/VAVBoxCoolingOnly.mo
+++ b/Buildings/Templates/ZoneEquipment/VAVBoxCoolingOnly.mo
@@ -5,6 +5,7 @@ model VAVBoxCoolingOnly "VAV terminal unit cooling only"
     redeclare final Buildings.Templates.Components.Coils.None coiHea);
 
   annotation (
+  __LinkageTemplate=true,
   defaultComponentName="VAVBox",
     Documentation(info="<html>
 <h4>Description</h4>
diff --git a/Buildings/Templates/ZoneEquipment/VAVBoxReheat.mo b/Buildings/Templates/ZoneEquipment/VAVBoxReheat.mo
index 05359d8bb0..f9e23f0735 100644
--- a/Buildings/Templates/ZoneEquipment/VAVBoxReheat.mo
+++ b/Buildings/Templates/ZoneEquipment/VAVBoxReheat.mo
@@ -4,6 +4,7 @@ model VAVBoxReheat "VAV terminal unit with reheat"
     final typ=Buildings.Templates.ZoneEquipment.Types.Configuration.VAVBoxReheat);
 
   annotation (
+   __LinkageTemplate=true,
    defaultComponentName="VAVBox",
     Documentation(info="<html>
 <h4>Description</h4>
diff --git a/Buildings/Templates/ZoneEquipment/package.mo b/Buildings/Templates/ZoneEquipment/package.mo
index ad01165130..8d4e3ffe19 100644
--- a/Buildings/Templates/ZoneEquipment/package.mo
+++ b/Buildings/Templates/ZoneEquipment/package.mo
@@ -2,7 +2,9 @@ within Buildings.Templates;
 package ZoneEquipment "Zone equipment"
   extends Modelica.Icons.Package;
 
-annotation (Documentation(info="<html>
+annotation (
+  __LinkageTemplate=true,
+  Documentation(info="<html>
 <p>
 This package contains templates for zone equipment (HVAC terminal units).
 </p>
