diff --git a/Makefile b/Makefile
index 26ab700..32db4c4 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ endif
 # download maven source file to current directory and change its name
 install-maven:
 	@echo "Installing maven"
-	curl https://dlcdn.apache.org/maven/maven-3/3.8.6/binaries/apache-maven-3.8.6-bin.tar.gz > apache-maven.tar.gz
+	curl -k https://dlcdn.apache.org/maven/maven-3/3.8.7/binaries/apache-maven-3.8.7-bin.tar.gz > apache-maven.tar.gz
 	mkdir -p apache_maven
 	tar xzf apache-maven.tar.gz -C apache_maven --strip-components 1
 	rm -rf apache-maven.tar.gz
